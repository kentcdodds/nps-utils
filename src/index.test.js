import path from 'path'

// all of these tests are run both as darwin and win32
// you provide the key (test name) and the value is a
// function that accepts npsUtils which you can use to
// create the result.
const snapshotTests = {
  series: ({series}) =>
    series('echo hey', null, 'echo hi', undefined, 'echo there', false),
  'series.nps': ({series}) =>
    series.nps(
      'test',
      false,
      'lint.src',
      null,
      'lint.scripts --cache',
      undefined,
      ' ',
      'build --fast',
    ),
  concurrent: ({concurrent}) => concurrent({
    test: 'echo test',
    validate: {
      script: false,
    },
    lint: {script: 'echo lint', color: 'bgWhite.black.dim'},
    build: false,
    cover: undefined,
  }),
  'concurrent.nps': ({concurrent}) =>
    concurrent.nps(
      null,
      'test',
      undefined,
      'lint',
      {script: 'build.app --silent'},
      false,
      {script: 'validate', color: 'bgGreen.dim'},
    ),
  runInNewWindow: ({runInNewWindow}) => runInNewWindow('echo hi'),
  'runInNewWindow.nps': ({runInNewWindow}) =>
    runInNewWindow.nps('lint --cache'),
  rimraf: ({rimraf}) => rimraf('build'),
  ifWindows: ({ifWindows}) => ifWindows('echo main', 'echo alternate'),
  ifNotWindows: ({ifNotWindows}) =>
    ifNotWindows('echo main', 'echo alternate'),
  copy: ({copy}) => copy('"**/*.html" "../dist/" --cwd=src --parents'),
  ncp: ({ncp}) => ncp('src dist'),
  mkdirp: ({mkdirp}) => mkdirp('/tmp/foo/bar/baz'),
  open: ({open}) =>
    open('http://kentcdodds.com -- "google chrome" --incognito'),
  crossEnv: ({crossEnv}) => crossEnv('NODE_ENV=test jest'),
  commonTags: ({commonTags}) =>
    commonTags.oneLine`  helpful\n  stuff\n  my good\n  friend`,
  setColors: ({concurrent, setColors}) => {
    setColors(['white.bgBlue.bold', 'black.bgYellow.dim'])
    return concurrent.nps('lint', 'build')
  },
  shellEscape: ({shellEscape}) =>
    shellEscape(`testing $:{}()[]ą^ęńł'÷/\\'\`" `),
  getBin: ({getBin}) => getBin('rimraf'),
}

Object.keys(snapshotTests).forEach(testName => {
  test(`${testName} as darwin`, () => {
    const result = withPlatform('darwin', snapshotTests[testName])
    expect(relativeizePath(result)).toMatchSnapshot()
  })
  test(`${testName} as win32`, () => {
    const result = withPlatform('win32', snapshotTests[testName])
    expect(relativeizePath(result)).toMatchSnapshot()
  })
})

function withPlatform(platform, getResult) {
  const originalPlatform = process.platform
  process.platform = platform
  jest.resetModules()
  const freshUtils = require('.')
  const result = getResult(freshUtils)
  process.platform = originalPlatform
  return result
}

function relativeizePath(stringWithAbsolutePaths) {
  // escape string for regexp generation
  const escapedPath = path
    .resolve(__dirname, '../')
    .replace(
      new RegExp('[\\-\\[\\]\\/\\{\\}\\(\\)\\*\\+\\?\\.\\\\^\\$\\|]', 'g'),
      '\\$&',
    )

  const relativePath = stringWithAbsolutePaths.replace(
    new RegExp(escapedPath, 'g'),
    '<projectRootDir>',
  )

  return relativePath.replace(/\\/g, '/')
}

/*
  eslint
    global-require:0,
    import/no-dynamic-require:0
*/
