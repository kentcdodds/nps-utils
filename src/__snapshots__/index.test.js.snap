// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commonTags as darwin 1`] = `"helpful stuff my good friend"`;

exports[`commonTags as win32 1`] = `"helpful stuff my good friend"`;

exports[`concurrent as darwin 1`] = `"node node_modules/concurrently/src/main.js --kill-others-on-fail --prefix-colors \\"bgBlue.bold,bgWhite.black.dim\\" --prefix \\"[{name}]\\" --names \\"test,lint\\" 'echo test' 'echo lint'"`;

exports[`concurrent as win32 1`] = `"node node_modules/concurrently/src/main.js --kill-others-on-fail --prefix-colors \\"bgBlue.bold,bgWhite.black.dim\\" --prefix \\"[{name}]\\" --names \\"test,lint\\" \\"echo test\\" \\"echo lint\\""`;

exports[`concurrent.nps as darwin 1`] = `"node node_modules/concurrently/src/main.js --kill-others-on-fail --prefix-colors \\"bgMagenta.bold,bgBlack.bold,bgCyan.bold,bgGreen.dim\\" --prefix \\"[{name}]\\" --names \\"test,lint,build.app,validate\\" 'nps test' 'nps lint' 'nps \\"build.app --silent\\"' 'nps validate'"`;

exports[`concurrent.nps as win32 1`] = `"node node_modules/concurrently/src/main.js --kill-others-on-fail --prefix-colors \\"bgMagenta.bold,bgBlack.bold,bgCyan.bold,bgGreen.dim\\" --prefix \\"[{name}]\\" --names \\"test,lint,build.app,validate\\" \\"nps test\\" \\"nps lint\\" \\"nps \\"\\"build.app --silent\\"\\"\\" \\"nps validate\\""`;

exports[`copy as darwin 1`] = `"node node_modules/cpy-cli/cli.js \\"**/*.html\\" \\"../dist/\\" --cwd=src --parents"`;

exports[`copy as win32 1`] = `"node node_modules/cpy-cli/cli.js \\"**/*.html\\" \\"../dist/\\" --cwd=src --parents"`;

exports[`crossEnv as darwin 1`] = `"node node_modules/cross-env/bin/cross-env.js NODE_ENV=test jest"`;

exports[`crossEnv as win32 1`] = `"node node_modules/cross-env/bin/cross-env.js NODE_ENV=test jest"`;

exports[`ifNotWindows as darwin 1`] = `"echo main"`;

exports[`ifNotWindows as win32 1`] = `"echo alternate"`;

exports[`ifWindows as darwin 1`] = `"echo alternate"`;

exports[`ifWindows as win32 1`] = `"echo main"`;

exports[`mkdirp as darwin 1`] = `"node node_modules/mkdirp/bin/cmd.js /tmp/foo/bar/baz"`;

exports[`mkdirp as win32 1`] = `"node node_modules/mkdirp/bin/cmd.js /tmp/foo/bar/baz"`;

exports[`ncp as darwin 1`] = `"node node_modules/ncp/bin/ncp src dist"`;

exports[`ncp as win32 1`] = `"node node_modules/ncp/bin/ncp src dist"`;

exports[`open as darwin 1`] = `"node node_modules/opn-cli/cli.js http://kentcdodds.com -- \\"google chrome\\" --incognito"`;

exports[`open as win32 1`] = `"node node_modules/opn-cli/cli.js http://kentcdodds.com -- \\"google chrome\\" --incognito"`;

exports[`rimraf as darwin 1`] = `"node node_modules/rimraf/bin.js build"`;

exports[`rimraf as win32 1`] = `"node node_modules/rimraf/bin.js build"`;

exports[`runInNewWindow as darwin 1`] = `"osascript -e 'tell application \\"Terminal\\"' -e 'tell application \\"System Events\\" to keystroke \\"t\\" using {command down}' -e 'do script \\"cd <projectRootDir> && echo hi\\" in front window' -e 'end tell'"`;

exports[`runInNewWindow as win32 1`] = `"start cmd /k \\"cd <projectRootDir> && echo hi\\""`;

exports[`runInNewWindow.nps as darwin 1`] = `"osascript -e 'tell application \\"Terminal\\"' -e 'tell application \\"System Events\\" to keystroke \\"t\\" using {command down}' -e 'do script \\"cd <projectRootDir> && node node_modules/.bin/nps /\\"lint --cache/\\"\\" in front window' -e 'end tell'"`;

exports[`runInNewWindow.nps as win32 1`] = `"start cmd /k \\"cd <projectRootDir> && node node_modules/.bin/nps /\\"lint --cache/\\"\\""`;

exports[`series as darwin 1`] = `"echo hey && echo hi && echo there"`;

exports[`series as win32 1`] = `"echo hey && echo hi && echo there"`;

exports[`series.nps as darwin 1`] = `"nps test && nps lint.src && nps \\"lint.scripts --cache\\" && nps \\"build --fast\\""`;

exports[`series.nps as win32 1`] = `"nps test && nps lint.src && nps \\"lint.scripts --cache\\" && nps \\"build --fast\\""`;

exports[`setColors as darwin 1`] = `"node node_modules/concurrently/src/main.js --kill-others-on-fail --prefix-colors \\"white.bgBlue.bold,black.bgYellow.dim\\" --prefix \\"[{name}]\\" --names \\"lint,build\\" 'nps lint' 'nps build'"`;

exports[`setColors as win32 1`] = `"node node_modules/concurrently/src/main.js --kill-others-on-fail --prefix-colors \\"white.bgBlue.bold,black.bgYellow.dim\\" --prefix \\"[{name}]\\" --names \\"lint,build\\" \\"nps lint\\" \\"nps build\\""`;
